core/src/Kokkos_Half.hpp

```diff
@@ -116,5 +113,7 @@ cast_from_half(half_t val) {
 }  // namespace Experimental
 }  // namespace Kokkos
 
-#endif
-#endif
+#else
+#define KOKKOS_HALF_T_IS_FLOAT false
```

Comment:
This might trip those who do `#ifdef KOKKOS_HALF_T_IS_FLOAT ` out of habit instead of `#if KOKKOS_HALF_T_IS_FLOAT`

Reply:
I'll defer to @crtrott on this. I can add a `KOKKOS_HALF_T_IS_NOT_FLOAT` define if that's preferred.

Reply:
so we either need the Macro as defined/undefined + the constexpr bool, or just the macro true or false.

Reply:
It was brought up at the developer meeting that misspelling the macro in an `#ifdef` is also a risk

Reply:
We decided this is fine for now